x/gamm/keeper/internal/events/emit.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+func EmitSwapEvent(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, input sdk.Coins, output sdk.Coins) {
+	if ctx.EventManager() == nil {
+		return
+	}
```

Comment:
asking to learn: what are the supposable scenario that we don't ctx doesn't have EventManager? We didn't have this check before right?